tests/unit/geometry.ts

```diff
@@ -0,0 +1,190 @@
+				width: 2 + 6 + 30 + 7 + 3
+			});
+		},
+
+		getContentBox() {
+			const box = geometry.getContentBox(element);
+			assert.deepEqual(box, {
+				bottom: 12 + 4 + 8 + 20,
+				height: 20,
+				left: 10 + 2 + 6,
+				right: 10 + 2 + 6 + 30,
+				top: 12 + 4 + 8,
+				width: 30
+			});
+		},
+
+		getMarginBox() {
+			const box = geometry.getMarginBox(element);
+			assert.deepEqual(box, {
+				bottom: 12 + 4 + 8 + 20 + 5 + 1 + 9,
+				height: 12 + 4 + 8 + 20 + 5 + 1 + 9,
+				left: 0,
+				right: 10 + 2 + 6 + 30 + 7 + 3 + 11,
+				top: 0,
+				width: 10 + 2 + 6 + 30 + 7 + 3 + 11
+			});
+		},
+
+		getPaddingBox() {
+			const box = geometry.getPaddingBox(element);
+			assert.deepEqual(box, {
+				bottom: 12 + 4 + 8 + 20 + 5,
+				height: 8 + 20 + 5,
+				left: 10 + 2,
+				right: 10 + 2 + 6 + 30 + 7,
+				top: 12 + 4,
+				width: 6 + 30 + 7
+			});
+		}
+	},
+
+	setters: (function () {
+		function testBoxMethod(method: string, boxSizing: string, expected: any) {
+			return function() {
+				element.style.boxSizing = boxSizing;
+				geometry[method](element, box);
+				const style = element.style;
+				assert.strictEqual(style.top, expected.top);
+				assert.strictEqual(style.left, expected.left);
+				assert.strictEqual(element.offsetWidth, expected.offsetWidth);
+				assert.strictEqual(element.offsetHeight, expected.offsetHeight);
+			}
+		}
+		return {
+			setup() {
+				box = {
+					top: 10,
+					left: 15,
+					width: 100,
+					height: 70
+				};
+			},
+			beforeEach: createTestElement(),
+
+			afterEach: destroyTestElement(),
+
+			setBorderBox: (function() {
+				let expected = {
+					top: '-2px',
+					left: '5px',
+					offsetWidth: 100,
+					offsetHeight: 70
+				};
+
+				return {
+					'box-sizing: border-box': testBoxMethod('setBorderBox', 'border-box', expected),
+					'box-sizing: content-box': testBoxMethod('setBorderBox', 'content-box', expected)
+				}
+			})(),
+
+			setContentBox: (function() {
+				let expected = {
+					top: '-14px',
+					left: '-3px',
+					offsetWidth: 118,
+					offsetHeight: 88
+				};
+
+				return {
+					'box-sizing: border-box': testBoxMethod('setContentBox', 'border-box', expected),
+					'box-sizing: content-box': testBoxMethod('setContentBox', 'content-box', expected)
+				}
+			})(),
+
+			setMarginBox: (function() {
+				let expected = {
+					top: '10px',
+					left: '15px',
+					offsetWidth: 79,
+					offsetHeight: 49
+				};
+
+				return {
+					'box-sizing: border-box': testBoxMethod('setMarginBox', 'border-box', expected),
+					'box-sizing: content-box': testBoxMethod('setMarginBox', 'content-box', expected)
+				}
+			})(),
+
+			setPaddingBox: (function() {
+				let expected = {
+					top: '-6px',
+					left: '3px',
+					offsetWidth: 105,
+					offsetHeight: 75
+				};
+
+				return {
+					'box-sizing: border-box': testBoxMethod('setPaddingBox', 'border-box', expected),
+					'box-sizing: content-box': testBoxMethod('setPaddingBox', 'content-box', expected)
+				}
+			})()
+		}
```

Comment:
We should add edge case tests:

* Passing only partial objects (i.e. not all 4 of `top`, `left`, `height`, and `width`)
* Passing negative numbers (which should also be ignored)
* Passing zero (which shouldn't technically be ignored - though I think Dojo 1's module would have ignored them for width/height)

It might be feasible to do this in such a way that we simply check for strict equality on things that shouldn't change, and strict inequality on things that should, in order to avoid a huge long-winded set of tests.